package.json

```diff
@@ -11,14 +11,15 @@
     "build:es": "rm -rf es && cross-env BABEL_ENV=es babel src -d es --extensions .ts,.tsx --ignore src/**/*.test.tsx",
     "build:ts": "tsc --rootDir src -d --declarationDir lib && tsc --rootDir src -d --declarationDir es",
     "build:icon": "rm -rf src/General/Icon/components && babel-node --extensions .ts,.tsx src/Utils/Icon/generateIcons.ts && npm run format:icon",
+    "check:branch": "if [ \"$(git rev-parse --abbrev-ref HEAD)\" = v4 ] ; then echo Branch check passed. ; else echo Error: Not on branch v4. ; exit 1 ; fi",
     "format:icon": "prettier --write src/General/Icon/components/*.tsx",
     "lint": "npm run lint:css && npm run lint:ts",
     "lint:css": "stylelint src/Style/**/*.ts",
     "lint:ts": "eslint src/**/*.{ts,tsx}",
-    "preversion": "npm whoami && npm run validate",
+    "preversion": "npm run check:branch && git pull && npm whoami && npm run validate",
     "postversion": "npm publish",
-    "prepublishOnly": "npm whoami && npm run validate",
-    "postpublish": "git push origin --tags",
+    "prepublishOnly": "npm run check:branch && git pull && npm whoami && npm run validate",
+    "postpublish": "git push && git push --tags",
```

Comment:
why we push `postpublish`

Reply:
you mean should do in `postversion`?

`postpublish` will can help us to ensure the publish is done, then we can do git push without worries about the git tag is in a wrong commit.

Reply:
I think we should push right after `npm version` so maybe postversion?

Reply:
Do you remember if the git tag confit occurs in `git push --tags`? If so, we should do it in `postversion`.

Reply:
maybe we should just have one standard way to do publish:
step 1. `npm version`
step 2. `npm publish`

Reply:
Yes `git push --tags` should not do in `postpublish` becuase those git change is made by `npm version`